src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -295,8 +296,12 @@ public static StandardCredentials lookupScanCredentials(
             @CheckForNull String repoOwner) {
         if (Util.fixEmpty(scanCredentialsId) == null) {
             return null;
+        }
+        StandardCredentials c = null;
+        if (scanCredentials.get() != null) {
```

Comment:
Would need to check that the `context`, `apiUri`, `scanCredentialsId`, `repoOwner` match the one previously looked up.. Maybe the combination of those parameters could make a hash to handle this. And callers should stop doing `Connector.lookupScanCredentials(context, apiUri, scanCredentialsId, repoOwner)` followed by `Connector.connect(apiUri, credentials)` but just a single call. And let the connector handle the lookup credentials if necessary.